File: leapYear.py
#!/usr/bin/python3
"""check if a year is a leap year"""


def leap():
    """leap year should be divisible by 4 or if century divisible by 400"""
    year = int(input("Enter the year : "))
    if (year % 4 == 0 and year % 100 != 0) or year % 400 == 0:
        print(f"{year} is a leap year")
    else:
        print(f"{year} is NOT a leap year")


if __name__ == "__main__":
    leap()

File: leapYear.java
import java.util.Scanner;

public class leapYear {
    public static void main(String[] args){
        Scanner myScanner = new Scanner(System.in);
        System.out.println("Enter the year: ");
        int year = myScanner.nextInt();
        myScanner.close();
        if ((year % 4 == 0 && year % 100 != 0) || year % 400 == 0){
            System.out.printf("The year %s is a leap year\n", year);
        }
        else{
            System.out.printf("Year %s is NOT a leap year\n", year);
        }
    }
}
